### PR TITLE
sort results of search index by matched count

### PR DESCRIPTION
**Pull request by javeme**
implement: #630

Change-Id: I6cab6741f2e0dfcdb731ffc099334508e33ca2e1

### CODE REVIEW

File: hugegraph-test/src/main/java/com/baidu/hugegraph/core/VertexCoreTest.java [hunk 1]
```diff
@@ -2448,6 +2448,85 @@ public void testQueryByTextContainsAndExactMatchProperty() {
...
+                        "lived", "San Francisco Bay Area");
+        graph.tx().commit();
+
+        List<Vertex> vertices = graph.traversal().V()
+                                     .hasLabel("author")
+                                     .has("lived", Text.contains("Bay Area"))
+                                     .has("~page", "").limit(2)
+                                     .toList();
+        Assert.assertEquals(2, vertices.size());
+        assertContains(vertices,
```
Comment (Linary) [inline hugegraph-test/src/main/java/com/baidu/hugegraph/core/VertexCoreTest.java]: this test can't see the in-page sorting effect

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/page/QueryList.java [hunk 2]
```diff
@@ -283,6 +291,12 @@ public boolean hasNextPage() {
                                  PageState.EMPTY_BYTES);
         }
 
+        public Query query() {
+            E.checkState(this.queries.size() == 1,
+                         "Unexpected queries size: %s", this.queries);
```
Comment (zhoney) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/backend/page/QueryList.java]: "Expect queries size 1, but got: %s"

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/GraphTransaction.java [hunk 3]
```diff
@@ -454,13 +455,19 @@ public void removeVertex(HugeVertex vertex) {
             ids.add(id);
         }
 
+        if (vertices.isEmpty() && !query.empty() &&
+            query.ids().size() == ids.size()) {
```
Comment (zhoney) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/GraphTransaction.java]: `query.ids().size() == ids.size()` implies `vertices.isEmpty()`

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/GraphTransaction.java [hunk 4]
```diff
@@ -589,13 +603,19 @@ public void removeEdge(HugeEdge edge) {
             ids.add(id);
         }
 
+        if (edges.isEmpty() && !query.empty() &&
+            query.ids().size() == ids.size()) {
```
Comment (zhoney) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/GraphTransaction.java]: ditto

File: hugegraph-test/src/main/java/com/baidu/hugegraph/core/VertexCoreTest.java (hunk 1 above)
Comment (javeme) [reply hugegraph-test/src/main/java/com/baidu/hugegraph/core/VertexCoreTest.java]: currently not support sorting in paging mode

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/page/PageEntryIterator.java [hunk 5]
```diff
@@ -106,4 +110,8 @@ public Object metadata(String meta, Object... args) {
         }
         throw new NotSupportException("Invalid meta '%s'", meta);
     }
+
+    public QueryResults results() {
```
[the remark this answers is not in the document]
Comment (javeme) [reply hugegraph-core/src/main/java/com/baidu/hugegraph/backend/page/PageEntryIterator.java]: renamed the PageIterator results

**Pull request merged**